docs/sources/reference/components/otelcol/otelcol.exporter.datadog.md

```diff
@@ -0,0 +1,324 @@
+---
```

Comment:
according to https://github.com/grafana/alloy/blob/main/docs/developer/adding-community-components.md, the documentation for community components should be in a different folder. I will let @clayton-cornell confirm whether we still want to do this